Source/JavaScriptCore/runtime/JSArrayBufferPrototype.cpp

```diff
@@ -239,12 +262,117 @@ JSC_DEFINE_HOST_FUNCTION(arrayBufferProtoFuncSlice, (JSGlobalObject* globalObjec
     return arrayBufferSlice(globalObject, callFrame->thisValue(), callFrame->argument(0), callFrame->argument(1), ArrayBufferSharingMode::Default);
 }
 
+JSC_DEFINE_HOST_FUNCTION(arrayBufferProtoFuncResize, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    // https://tc39.es/proposal-resizablearraybuffer/#sec-arraybuffer.prototype.resize
+
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSValue arrayBufferValue = callFrame->thisValue();
+
+    JSArrayBuffer* thisObject = jsDynamicCast<JSArrayBuffer*>(arrayBufferValue);
+    if (!thisObject || (ArrayBufferSharingMode::Shared == thisObject->impl()->sharingMode()))
+        return throwVMTypeError(globalObject, scope, "Receiver must be ArrayBuffer"_s);
+
+    if (UNLIKELY(!thisObject->impl()->isResizableOrGrowableShared()))
+        return throwVMTypeError(globalObject, scope, "ArrayBuffer is not resizable"_s);
+
+    if (UNLIKELY(thisObject->impl()->isDetached()))
+        return throwVMTypeError(globalObject, scope, "Receiver is detached"_s);
+
+    double newLength = callFrame->argument(0).toIntegerOrInfinity(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    if (!std::isfinite(newLength) || newLength < 0)
+        return throwVMRangeError(globalObject, scope, "new length is out of range"_s);
+    size_t newByteLength = static_cast<size_t>(newLength);
```

Comment:
nit: I always feel like `auto ... = static_cast<T>` is less redundant than `T ... = static_cast<T>`, but it really doesn't matter.

Reply:
OK